backend/app/Repositories/EloquentChartVisitorsRepository.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Actions\Visitors\GetNewChartVisitorsByDateRangeRequest;
+use App\Repositories\Contracts\ChartVisitorsRepository;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class EloquentChartVisitorsRepository implements ChartVisitorsRepository
+{
+    public function getNewVisitorsByDate(GetNewChartVisitorsByDateRangeRequest $request): Collection
+    {
+        $subQueryFirst = "SELECT id FROM websites where user_id = :user_id";
+        $subQuerySecond = "SELECT visitors.*, ( " .$this->getPeriod('created_at', $request->getPeriod()) .") as period
+                             FROM \"visitors\" WHERE
+                                      created_at >= :startData and created_at <= :endData and website_id = ($subQueryFirst)";
+        $query = DB::raw("SELECT COUNT(*), period from ($subQuerySecond) as periods group by period");
+        $response = DB::select((string)$query, [
+            'startData' => Carbon::createFromTimestampUTC($request->getStartDate())->toDateTimeString(),
+            'endData' => Carbon::createFromTimestampUTC($request->getEndDate())->toDateTimeString(),
+            'user_id' => Auth::user()->id
```

Comment:
repository mustn't know anything about authorized user, move getting user to action, and pass user id as parameter